tensorboard/uploader/uploader_main.py

```diff
@@ -439,6 +463,26 @@ def _get_intent(flags):
     raise AssertionError('Unknown subcommand %r' % (cmd,))
 
 
+def _get_server_info(flags):
+  if flags.api_endpoint:
+    return server_info_lib.create_server_info(flags.origin, flags.api_endpoint)
+  else:
+    return server_info_lib.fetch_server_info(flags.origin)
+
+
+def _handle_server_info(info):
+  compat = info.compatibility
+  if compat.verdict == server_info_pb2.VERDICT_WARN:
+    sys.stderr.write('Warning: %s\n' % compat.details)
+    sys.stderr.flush()
+  elif compat.verdict == server_info_pb2.VERDICT_ERROR:
+    _die('Error: %s' % compat.details)
+  else:
+    # OK or unknown; assume OK.
+    if compat.details:
+      sys.stderr.write('%s\n' % compat.details)
```

Comment:
sys.stderr.flush() too?

Though now that I actually go look at the docs, apparently both it and stdout are supposed to be line-buffered, so maybe the fact that I usually flush() after doing a raw write() is just cargo culting?

Reply:
Fascinating. I had always assumed that stderr was unbuffered, like it is
in every other environment, because that’s one of the main points of
stderr. And so it was in Python 2, but indeed this appears to have
changed in Python 3. I’ll keep this in mind from now on; thanks.

(It’s not clear to me whether we _should_ flush here: superfluous in the
interactive case, but not when outputting to a file. I guess the perf
isn’t really a problem since this isn’t in a loop, so I’ll go ahead and
add it.)